### ISSUE TITLE
[Scooby-Doo: Curse of Anubis - Pyramid of Doom!] The game cannot proceed because the character is stuck

### ISSUE THREAD

**Issue by Jmousy** [ruffle-rs/ruffle#5926]
### Describe the bug

![scooby-doo](https://user-images.githubusercontent.com/75052171/147377904-3943f593-bd0f-4511-9ef0-31c036803139.gif)

When you start the game for the first time in **'Scooby-Doo: Curse of Anubis - Pyramid of Doom!'**, your character moves forward.
However, the Ruffle just keeps walking in place and the keyboard input is stuck.

The gif above is what happens when do nothing.

Ruffle Nightly 2021-12-22

[Scooby-Doo-Curse-of-Anubis-Pyramid-of-Doom.zip](https://github.com/ruffle-rs/ruffle/files/7775489/Scooby-Doo-Curse-of-Anubis-Pyramid-of-Doom.zip)

### Expected behavior

The character should initially move forward one step and then move by keyboard input.

### Affected platform

Self-hosted version

### Operating system

Windows 10 21H2 x64

### Browser

Google Chrome 96

**Comment by Toad06** [comment]
Thanks for the report.

The issue is in `\DefineSprite (930)\frame 1\PlaceObject2 (745) Depth: 380 (Pers)\onClipEvent(enterFrame)`:

```
onClipEvent(enterFrame){
   if(_global.C[Nom].CanMove == true)
   {
      _global.C[Nom].Move();
   }
}
```
The variable `Nom` doesn't hold the same value in Flash Player and Ruffle:
- "Pers" in Flash.
- "undefined" in Ruffle.

---

The variable `Nom` is defined in `\DefineSprite (930)\frame 1\PlaceObject2 (745) Depth: 380 (Pers)\onClipEvent(load)`:
```
onClipEvent(load){
   with(_parent._parent._parent)
   {
      var Nom = this._name
      ...
```

Adding a line `var Nom = this._name;` outside of the `with` statement appears to be a workaround for Ruffle:
```
onClipEvent(load){
   var Nom = this._name;
   with(_parent._parent._parent)
   {
      var Nom = this._name
      ...
```

**Comment by Toad06** [comment]
Fixed by #8953
